File: LeetCode/70. Climbing Stairs.java
class Solution {
    public int climbStairs(int n) {
        int[] dp = new int[n + 2];
        dp[0] = 1;
        for (int i = 0; i < n; i++) {
            dp[i+1] += dp[i];
            dp[i+2] += dp[i];
        }
        return dp[n];
    }
} //TC: O(n), SC: O(n)

File: LeetCode/70. Climbing Stairs.kt
class Solution {
    fun climbStairs(n: Int): Int {
        //by calculating ways to n backwards
        var prev = 1 //ways from the previous one stair to the last stair is always 1
        var next = 1 //ways to the last stair is always 1

        for (i in 0 until n-1) { //loop each stair for ways to there
            val tmp = prev
            prev = prev+next //adding up sets of ways to next 2 stairs
            next = tmp //next is now the old prev
        }

        return prev
    }
}
